app/react-native/src/preview/components/OnDeviceUI/index.js

```diff
@@ -1,24 +1,116 @@
-import React, { PropTypes } from 'react';
-import { View } from 'react-native';
+import React, { Component, PropTypes } from 'react';
+import { Animated, Easing, View, TouchableWithoutFeedback, Image, Text } from 'react-native';
 import style from './style';
 import StoryListView from '../StoryListView';
 import StoryView from '../StoryView';
 
-export default function OnDeviceUI(props) {
-  const { stories, events, url } = props;
+export default class OnDeviceUI extends Component {
+  constructor(props, ...args) {
+    super(props, ...args);
 
-  return (
-    <View style={style.main}>
-      <View style={style.leftPanel}>
-        <StoryListView stories={stories} events={events} />
-      </View>
-      <View style={style.rightPanel}>
-        <View style={style.preview}>
-          <StoryView url={url} events={events} />
+    this.state = {
+      menuAnimation: new Animated.Value(0),
+      isMenuOpen: false,
+      selectedKind: null,
+      selectedStory: null,
+    };
+
+    this.storyChangedHandler = this.handleStoryChanged.bind(this);
+    this.handleToggleMenu = this.handleToggleMenu.bind(this);
+
+    this.props.events.on('story', this.storyChangedHandler);
+  }
+
+  componentWillUnmount() {
+    this.props.events.removeListener('story', this.storyChangedHandler);
+  }
+
+  handleStoryChanged(storyFn, selection) {
+    const { kind, story } = selection;
+    this.setState({
+      selectedKind: kind,
+      selectedStory: story,
+    });
+  }
+
+  handleToggleMenu() {
+    const isMenuOpen = !this.state.isMenuOpen;
+
+    Animated.timing(this.state.menuAnimation, {
+      toValue: isMenuOpen ? 1 : 0,
+      duration: 150,
+      easing: Easing.linear,
+    }).start();
+
+    this.setState({
+      isMenuOpen,
+    });
+  }
+
+  render() {
+    const { stories, events, url } = this.props;
```

Comment:
The original component received added stories through event emissions -- I think we may want to subscribe to those here too, or otherwise the sidebar may not update as stories become added. cc @shilman or @ndelangen -- do Preview components receive additional stories on `props.stories` over time? Or are subscriptions needed